### PR TITLE
Update some links and docs

### LINKED ISSUE #1342: New git hook script name
https://github.com/rust-lang/rustc-dev-guide/blob/master/src/building/suggested.md contains broken link `https://github.com/rust-lang/rust/blob/master/src/etc/pre-commit.sh`. This script is now called [`pre-push.sh`](https://github.com/rust-lang/rust/blob/master/src/etc/pre-push.sh)

### PR DESCRIPTION
**Pull request by JohnTitor**
Fixes some links while updating outdated docs.
Closes #1342

### CODE REVIEW

File: src/query.md [hunk 1]
```diff
@@ -37,7 +37,7 @@ will in turn demand information about that crate, starting from the
   actual parsing.
 
 Although this vision is not fully realized, large sections of the
-compiler (for example, generating [MIR](./mir/)) currently work exactly like this.
+compiler (for example, generating [MIR](./mir/index.md)) currently work exactly like this.
```
Comment (JohnTitor) [inline src/query.md:40]: This isn't actually broken but the linkchecker complains about it.

File: src/tests/running.md [hunk 2]
```diff
@@ -211,7 +211,7 @@ See [Compare modes](compiletest.md#compare-modes) for more details.
 
 Sometimes it's easier and faster to just run the test by hand.
 Most tests are just `rs` files, so after
-[creating a rustup toolchain](/building/how-to-build-and-run.html#creating-a-rustup-toolchain),
+[creating a rustup toolchain](../building/how-to-build-and-run.md#creating-a-rustup-toolchain),
```
Comment (JohnTitor) [inline src/tests/running.md:214]: Same here

File: src/macro-expansion.md [hunk 3]
```diff
@@ -515,20 +516,20 @@ We use these items in macro parser:
   ask the MBE parser to parse. We will consume the raw stream of tokens and
   output a binding of metavariables to corresponding token trees. The parsing
   session can be used to report parser errors.
-- `ms` a _matcher_. This is a sequence of token trees that we want to match
-  the token stream against.
+- `matcher` is a sequence of `MatcherLoc`s that we want to match
+  the token stream against. They're converted from token trees before matching.
```
Comment (JohnTitor) [inline src/macro-expansion.md:520]: Not sure the updated doc is correct, cc @petrochenkov could you review if you have some time?

**Comment by JohnTitor** [comment]
@rust-lang/wg-rustc-dev-guide It'd be great if someone could review this :)

File: src/macro-expansion.md (hunk 3 above)
Comment (spastorino) [reply src/macro-expansion.md:520]: I'm not sure about this part either.

Comment (petrochenkov) [reply src/macro-expansion.md:520]: Looks correct.

**Comment by JohnTitor** [comment]
Thanks both for reviewing! Merging.

**Pull request merged**